src/sentry/integrations/github/integration.py

```diff
@@ -900,8 +962,85 @@ def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase
                 lifecycle.record_failure(GitHubInstallationError.USER_MISMATCH)
                 return error(
                     request,
-                    self.active_organization,
+                    self.active_user_organization,
                     error_short=GitHubInstallationError.USER_MISMATCH,
                 )
 
             return pipeline.next_step()
+
+    def check_pending_integration_deletion(self, request: HttpRequest) -> HttpResponse | None:
+        if self.active_user_organization is None:
+            return None
+        # We want to wait until the scheduled deletions finish or else the
+        # post install to migrate repos do not work.
+        integration_pending_deletion_exists = OrganizationIntegration.objects.filter(
+            integration__provider=GitHubIntegrationProvider.key,
+            organization_id=self.active_user_organization.organization.id,
+            status=ObjectStatus.PENDING_DELETION,
+        ).exists()
+
+        if integration_pending_deletion_exists:
+            return error(
+                request,
+                self.active_user_organization,
+                error_short=GitHubInstallationError.PENDING_DELETION,
+                error_long=ERR_INTEGRATION_PENDING_DELETION,
+            )
+        return None
+
+
+class GithubOrganizationSelection(PipelineView):
+    def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase:
+        self.active_user_organization = determine_active_organization(request)
+
+        if self.active_user_organization is None or not features.has(
+            "organizations:github-multi-org",
+            organization=self.active_user_organization.organization,
+            actor=request.user,
+        ):
+            return pipeline.next_step()
+
+        with record_event(
+            IntegrationPipelineViewType.ORGANIZATION_SELECTION
+        ).capture() as lifecycle:
+            installation_info = pipeline.fetch_state("existing_installation_info") or []
+            if len(installation_info) == 0:
+                return pipeline.next_step()
+
+            # add an option for users to install on a new GH organization
+            installation_info.append(
+                {
+                    "installation_id": "-1",
+                    "github_account": "Install integration on a new GitHub organization",
+                    "avatar_url": "https://raw.githubusercontent.com/getsentry/sentry/526f08eeaafa3a830f70671ad473afd7b9b05a0f/src/sentry/static/sentry/images/logos/sentry-avatar.png",
```

Comment:
This avatar url doesn't really matter since we will use [a + icon ](https://github.com/getsentry/sentry/pull/91381)instead